ArcGISRuntimeSDKQt_CppSamples/Scenes/FilterFeaturesInScene/FilterFeaturesInScene.cpp

```diff
@@ -0,0 +1,168 @@
+  basemap->baseLayers()->append(m_osmBuildings);
+  m_scene = new Scene(basemap, this);
+
+  // Construct and set the scene topography
+  Surface* surface = new Surface(this);
+  surface->elevationSources()->append(new ArcGISTiledElevationSource(QUrl("https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"), this));
+  m_scene->setBaseSurface(surface);
+
+  // Construct the detailed buildings scene layer
+  m_detailedBuildingsSceneLayer = new ArcGISSceneLayer(QUrl("https://tiles.arcgis.com/tiles/z2tnIkrLQ2BRzr6P/arcgis/rest/services/SanFrancisco_Bldgs/SceneServer"), this);
+
+  connect(m_detailedBuildingsSceneLayer, &ArcGISSceneLayer::doneLoading, this, [this](const Error error)
+  {
+    if (!error.isEmpty())
+    {
+      qWarning() << "San Fransisco Buildings Layer failed to load." << error.message() << error.additionalMessage();
+      return;
+    }
+
+    // Construct a red polygon that shows the extent of the detailed buildings scene layer
+    Envelope sfExtent = m_detailedBuildingsSceneLayer->fullExtent();
+
+    PolygonBuilder* builder = new PolygonBuilder(m_sceneView->spatialReference(), this);
+    builder->addPoint(sfExtent.xMin(), sfExtent.yMin());
+    builder->addPoint(sfExtent.xMax(), sfExtent.yMin());
+    builder->addPoint(sfExtent.xMax(), sfExtent.yMax());
+    builder->addPoint(sfExtent.xMin(), sfExtent.yMax());
+
+    m_sceneLayerExtentPolygon = builder->toPolygon();
+
+    // Create the SceneLayerPolygonFilter to later apply to the OSM buildings layer
+    m_sceneLayerPolygonFilter = new SceneLayerPolygonFilter(
+          QList<Polygon>{m_sceneLayerExtentPolygon}, // polygons to filter by
+          SceneLayerPolygonFilterSpatialRelationship::Disjoint, // hide all features within the polygons
+          this);
+
+    // Create the extent graphic so we can add it later with the detailed buildings scene layer
+    SimpleFillSymbol* sfs = new SimpleFillSymbol(SimpleFillSymbolStyle::Solid, QColor(Qt::transparent), new SimpleLineSymbol(SimpleLineSymbolStyle::Solid, QColor(Qt::red), 5.0f, this), this);
+    m_sanFransiscoExtentGraphic = new Graphic(m_sceneLayerExtentPolygon, sfs, this);
+  });
+
+  m_detailedBuildingsSceneLayer->load();
+}
+
+FilterFeaturesInScene::~FilterFeaturesInScene() = default;
+
+void FilterFeaturesInScene::init()
+{
+  // Register the scene view for QML
+  qmlRegisterType<SceneQuickView>("Esri.Samples", 1, 0, "SceneView");
+  qmlRegisterType<FilterFeaturesInScene>("Esri.Samples", 1, 0, "FilterFeaturesInSceneSample");
+}
+
+void FilterFeaturesInScene::loadScene()
+{
+  // Show the detailed buildings scene layer and the extent graphic
+  m_scene->operationalLayers()->append(m_detailedBuildingsSceneLayer);
+  m_sceneView->graphicsOverlays()->first()->graphics()->append(m_sanFransiscoExtentGraphic);
+}
+
+void FilterFeaturesInScene::filterScene()
+{
+  // Hide buildings within the detailed building extent so they don't clip
```

Comment:
what code is this comment referring to?

Reply:
It's referring to the code that occurs in the method - either the `if` or `else` blocks. It's intended to be kind of like a header. We don't normally put comments before the method (ie line 120) so I placed it there.